sdk/storage/azure-storage-blob/azure/storage/blob/_shared_access_signature.py

```diff
@@ -308,15 +308,17 @@ def get_token(self):
 
 
 def generate_account_sas(
-        account_name,  # type: str
-        account_key,  # type: str
-        resource_types,  # type: Union[ResourceTypes, str]
-        permission,  # type: Union[AccountSasPermissions, str]
-        expiry,  # type: Union[datetime, str]
-        start=None,  # type: Optional[Union[datetime, str]]
-        ip=None,  # type: Optional[str]
-        **kwargs # type: Any
-    ):  # type: (...) -> str
+    account_name: str,
+    account_key: str,
+    resource_types: Union["ResourceTypes", str],
+    permission: Union["AccountSasPermissions", str],
+    expiry: Union["datetime", str],
+    start: Optional[Union["datetime", str]] = None,
+    ip: Optional[str] = None,
+    *,
+    services: Services = Services(blob=True),
```

Comment:
Does the underlying logic also accept a `str`? If so, we should type it as such but still use `Services` as the default.

Reply:
Nope, passing `services='blob'` yields a
```
<AuthenticationErrorDetail>Signature fields not well formed.</AuthenticationErrorDetail></Error>

<string>:1: ClientAuthenticationError
```

Reply:
`'blob`' would not be the right string to pass. You would want to match the `ss` param from [here](https://learn.microsoft.com/en-us/rest/api/storageservices/create-account-sas#specify-the-account-sas-parameters). So just `'b'` for Blob or `'bf'` for Blob and File together.

The fact that it made it to the service makes me think we do indeed take a string and pass it through. Give that a try and confirm by looking at the internal code.

Reply:
Good callout, will add typehint for strings 😄